x/payment/keeper/keeper.go

```diff
@@ -32,7 +41,8 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 func (k Keeper) PayForData(goCtx context.Context, msg *types.MsgPayForData) (*types.MsgPayForDataResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	ctx.GasMeter().ConsumeGas(msg.MessageSize, payForDataGasDescriptor)
+	gasToConsume := uint64(shares.MsgSharesUsed(int(msg.MessageSize)) * GasPerMsgShare)
```

Comment:
[Non Blocking][Question]
From the gas docs:
> The AnteHandler is run for every transaction during CheckTx and DeliverTx, before a Protobuf Msg service method for each sdk.Msg in the transaction.

So, users will pay `gasToConsume` + `checkTx|DeliverTx gas` ?  But, since we're not touching any state, the `checkTx|DeliverTx gas` will be 0 ?

Reply:
> But, since we're not touching any state, the checkTx|DeliverTx gas will be 0 ?

I think so

Reply:
the gas for checktx and delivertx will still have the normal sig verification and standard 10 gas/byte of tx costs as well